File: server-php/src/Controllers/TagsController.php
<?php

declare(strict_types=1);

namespace EnklWiki\Controllers;

use EnklWiki\Request;
use EnklWiki\Response;
use EnklWiki\Services\TagService;

final class TagsController
{
    public function __construct(private readonly TagService $tags)
    {
    }

    public function deleteUnused(Request $request): Response
    {
        $removed = $this->tags->removeUnusedTags();
        return Response::json(['removed' => $removed]);
    }
}

File: server/EnklWiki.Api/Controllers/TagsController.cs
using EnklWiki.Api.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace EnklWiki.Api.Controllers;

[ApiController]
[Route("api/tags")]
// Tag pruning is only ever invoked from Site Settings on the client.
[Authorize(Roles = "admin")]
public class TagsController(TagService tagService) : ControllerBase
{
    [HttpDelete("unused")]
    public async Task<IActionResult> DeleteUnused()
    {
        var removed = await tagService.RemoveUnusedTagsAsync();
        return Ok(new { removed });
    }
}
